src/responder/ifp/ifp_users.c

```diff
@@ -1029,6 +1031,320 @@ ifp_users_list_by_domain_and_name_recv(TALLOC_CTX *mem_ctx,
+                                  req, &state->child_ctx);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE, "Could not set up child handlers [%d]: %s\n",
+                  ret, sss_strerror(ret));
+            ret = ERR_P11_CHILD;
+            goto done;
+        }
+
+        /* Set up timeout handler */
+        tv = tevent_timeval_current_ofs(state->timeout, 0);
+        endtime = tevent_req_set_endtime(req, state->ev, tv);
+        if (endtime == false) {
+            ret = ERR_P11_CHILD;
+            goto done;
+        }
+        /* Now either wait for the timeout to fire or the child to finish */
+    } else { /* error */
+        ret = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE, "fork failed [%d][%s].\n",
+              ret, sss_strerror(ret));
+        goto done;
+    }
+
+    return EAGAIN;
+
+done:
+    if (ret != EOK) {
+        PIPE_CLOSE(pipefd_from_child);
+        PIPE_CLOSE(pipefd_to_child);
+    }
+
+    return ret;
+}
+
+static void
+ifp_users_find_by_valid_cert_done(int child_status,
+                                  struct tevent_signal *sige,
+                                  void *pvt)
+{
+    struct tevent_req *subreq;
+    struct tevent_req *req = talloc_get_type(pvt, struct tevent_req);
+    struct ifp_users_find_by_valid_cert_state *state;
+    errno_t ret;
+
+    state = tevent_req_data(req, struct ifp_users_find_by_valid_cert_state);
+
+    PIPE_FD_CLOSE(state->io->read_from_child_fd);
+    PIPE_FD_CLOSE(state->io->write_to_child_fd);
+
+    if (WIFEXITED(child_status)) {
+        if (WEXITSTATUS(child_status) != 0) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  P11_CHILD_PATH " failed with status [%d]\n", child_status);
+            tevent_req_error(req, ERR_INVALID_CERT);
+            return;
+        }
+    } else if (WIFSIGNALED(child_status)) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              P11_CHILD_PATH " was terminated by signal [%d]\n",
+              WTERMSIG(child_status));
+        tevent_req_error(req, ECHILD);
+        return;
+    }
+
+    DEBUG(SSSDBG_TRACE_LIBS, "Certificate [%s] is valid.\n",
+          state->extra_args[0]);
+
+    subreq = cache_req_user_by_cert_send(state, state->ifp_ctx->rctx->ev,
+                                            state->ifp_ctx->rctx,
+                                            state->ifp_ctx->rctx->ncache, 0,
+                                            CACHE_REQ_ANY_DOM, NULL, state->derb64);
+    if (subreq == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Unable to create subrequest!\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    tevent_req_set_callback(subreq, ifp_users_find_by_valid_cert_step, req);
+
+    ret = EAGAIN;
+
+done:
+    if (ret != EAGAIN) {
+        tevent_req_error(req, ret);
+        tevent_req_post(req, state->ifp_ctx->rctx->ev);
+    }
+
+    return;
+}
+
+static void ifp_users_find_by_valid_cert_step(struct tevent_req *subreq)
```

Comment:
The last function in tevent request flow must be called request_done. This must be called `ifp_users_find_by_valid_cert_done`